lib/sql.js

```diff
@@ -156,200 +228,871 @@ SqlConnector.prototype.tableEscaped = function (model) {
+ */
+SqlConnector.prototype.deleteAll = function(model, where, options, cb) {
+  var self = this;
+  var stmt = this.buildDelete(model, where, options);
+  this.execute(stmt.sql, stmt.params, options, function(err, info) {
+    var affectedRows = self.getCountForAffectedRows(model, info);
+    if (cb) {
+      cb(err, {count: affectedRows});
+    }
   });
 };
+SqlConnector.prototype.destroyAll = SqlConnector.prototype.deleteAll;
 
 /**
- * Delete a model instance by id value
+ * Update attributes for a given model instance
  * @param {String} model The model name
  * @param {*} id The id value
- * @param {Function} callback The callback function
+ * @param {Object} data The model data instance containing all properties to
+ * be updated
+ * @param {Object} options Options object
+ * @param {Function} cb The callback function
  */
-SqlConnector.prototype.delete =
-SqlConnector.prototype.destroy = function destroy(model, id, callback) {
-  if (!isIdValuePresent(id, callback, true)) {
+SqlConnector.prototype.updateAttributes = function(model, id, data, options, cb) {
+  if (!isIdValuePresent(id, cb)) {
     return;
   }
-  var sql = 'DELETE FROM ' + this.tableEscaped(model) + ' WHERE ' +
-    this.idColumnEscaped(model) + ' = ' + this._escapeIdValue(model, id);
+  var idName = this.idName(model);
+  delete data[idName];
+  var where = {};
+  where[idName] = id;
+  this.updateAll(model, where, data, options, cb);
+};
 
-  this.command(sql, function (err, result) {
-    if (callback) callback(err, result);
+/**
+ * Build the UPDATE statement
+ * @param {String} model The model name
+ * @param {Object} where The where object
+ * @param {Object} data The data to be changed
+ * @param {Object} options The options object
+ * @param {Function} cb The callback function
+ * @returns {ParameterizedSQL} The UPDATE SQL statement
+ */
+SqlConnector.prototype.buildUpdate = function(model, where, data, options) {
+  var fields = this.buildFieldsForUpdate(model, data);
+  var updateClause = new ParameterizedSQL('UPDATE ' + this.tableEscaped(model));
+  var whereClause = this.buildWhere(model, where);
+  updateClause.merge([fields, whereClause]);
+  return this.parameterize(updateClause);
+};
+
+/**
+ * Update all instances that match the where clause with the given data
+ * @param {String} model The model name
+ * @param {Object} where The where object
+ * @param {Object} data The property/value object representing changes
+ * to be made
+ * @param {Object} options The options object
+ * @param {Function} cb The callback function
+ */
+SqlConnector.prototype.updateAll = function(model, where, data, options, cb) {
+  var self = this;
+  var stmt = this.buildUpdate(model, where, data, options);
+  this.execute(stmt.sql, stmt.params, options, function(err, info) {
+    var affectedRows = self.getCountForAffectedRows(model, info);
+    if (cb) {
+      cb(err, {count: affectedRows});
+    }
   });
 };
+SqlConnector.prototype.update = SqlConnector.prototype.updateAll;
```

Comment:
why do we have aliases... I wish we could get rid of these to make it less confusing when implementing a connector